src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -436,6 +437,7 @@ def select(self, cols: List[Expression]) -> "SelectStatement":
             len(cols) == 1
             and isinstance(cols[0], UnresolvedAlias)
             and isinstance(cols[0].child, Star)
+            and not cols[0].child.expressions
```

Comment:
when will the child has expressions?

Reply:
A simple example is that `df.select(df["*"])` has child expressions, which are attribute list of all columns from `df`.
When joining two dataframes, you can do the following. 
```python
    df_left = session.create_dataframe([[1, 2]]).to_df("a", "b")
    df_right = session.create_dataframe([[3, 4]]).to_df("c", "d")
    df = df_left.join(df_right)
    df = df.select(df_right["*"], df_left["*"])  # the two stars have child expressions.
    # the final df has columns C D B A instead of A B C D
```
`df.select("*")` doesn't have the child expressions.